Volume/Views/WeeklyDebriefView.swift

```diff
@@ -6,112 +6,251 @@
+                    .padding([.leading, .trailing], 50)
                 }
-                .padding([.leading, .trailing], 50)
+                .tabViewStyle(PageTabViewStyle())
             }
-            .tabViewStyle(PageTabViewStyle())
         }
-        .onAppear() {
+        .onAppear {
             UIPageControl.appearance().currentPageIndicatorTintColor = .gray
             UIPageControl.appearance().pageIndicatorTintColor = .lightGray
+            
+            if let uuid = userData.uuid {
+                fetchWeeklyDebrief(for: uuid)
+            }
+            
         }
-        .onDisappear() {
+        .onDisappear {
             UIPageControl.appearance().currentPageIndicatorTintColor = nil
             UIPageControl.appearance().pageIndicatorTintColor = nil
         }
     }
+    
+    
+    // MARK: - Copying from BrowserView
+    
+    private func fetchWeeklyDebrief(for uuid: String) {
+        state = .loading
+        cancellableArticleQuery = Network.shared.publisher(for: GetWeeklyDebriefQuery(uuid: uuid))
+            .sink { completion in
+                if case let .failure(_) = completion {
+                    // We are assuming this case won't occur
+                    print("Debrief Requested but not Acquired")
+                }
+            }
+            receiveValue: { debriefData in
+                let debrief = WeeklyDebrief(from: debriefData.user.weeklyDebrief)
+                fetchInfoFor(debrief: debrief)
+            }
+    }
+    
+    private func fetchInfoFor(debrief: WeeklyDebrief) {
+        
+        var debriefDisplayInfo = WeeklyDebriefDisplayInfo(
+            creationDate: debrief.creationDate,
+            expirationDate: debrief.expirationDate,
+            numShoutouts: debrief.numShoutouts,
+            numReadArticles: debrief.numReadArticles,
+            numBookmarkedArticles: debrief.numBookmarkedArticles,
+            readArticles: [],
+            randomArticles: [])
+        
+        func asyncPopulateDisplayInfo(recievedArticles: Int = 0, ids: [ArticleID], readNotRandom: Bool, completion: @escaping () -> ()) {
+            if recievedArticles == ids.count {
+                completion()
+                return
+            }
+            
+            cancellableArticleQuery = Network.shared.publisher(for: GetArticleByIdQuery(id: ids[recievedArticles]))
+                .sink { completion in
+                    if case let .failure(error) = completion {
+                        print(error.localizedDescription)
+                    }
+                }
+                receiveValue: { articleFields in
+                    let article = Article(from: articleFields.article!.fragments.articleFields)
+                    
+                    if readNotRandom {
+                        debriefDisplayInfo.readArticles.append(article)
+                    } else {
+                        debriefDisplayInfo.randomArticles.append(article)
+                    }
```

Comment:
Stylistic change–I'd ask Han
```suggestion
                    (readNotRandom ? debriefDisplayInfo.readArticles : debriefDisplayInfo.randomArticles).append(article)
```